tests/neg/OrType.scala

```diff
@@ -2,7 +2,7 @@ class B(val x: Int)
 class C(val x: Double)
 
 object Test{
-  def bar(x: B | C): Int | Double = x.x
+  def bar(x: B | C): Int | Double = x.x  // error
```

Comment:
I also assume that there might be nice use cases when interfacing with JS APIs, https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#union-types has some examples.